src/stream_base.h

```diff
@@ -214,6 +214,13 @@ class StreamResource {
   // Return 0 for success and a libuv error code for failures.
   virtual int DoTryWrite(uv_buf_t** bufs, size_t* count);
   // Perform a write of data, and either call req_wrap->Done() when finished
+  //   XXX call when? can we call sync? "call Done() and return" suggests first
+  //   Done(), then return, but that fails with TLSWrap.
+  //   See writeGeneric() in stream_base_commons.js:
+  //      What happens if Done() is called before return is that onwritecomplete
+  //      gets called sync inside handlWriteReq(), BEFORE afterWriteDispatched()
+  //      is called, but that means it happens before afterWriteDispatched()
+  //      sets req.callback = cb, so the cb never gets called.
```

Comment:
You're right, this currently only handles async completions. I assume we could do something like we do for `DoShutdown()`, that is, use a return code of `1` for synchronous success?

Reply:
I guess that could work, I guess that positive values aren't meaningful now? They aren't the amount of data written?

Reply:
Yeah, they should currently be unused, we only return 0 or a libuv error code atm.